cli/src/command/account.rs

```diff
@@ -110,3 +161,106 @@ impl CommandT for Transfer {
         Ok(())
     }
 }
+
+/// Accounts Storage
+///
+/// Manages [Account]s stored in the filesystem,
+/// providing ways to store and retrieve them.
+pub mod storage {
+    use directories::BaseDirs;
+    use serde_json as json;
+    use sp_core::serde::{Deserialize, Serialize};
+    use std::collections::HashMap;
+    use std::fs::File;
+    use std::fs::OpenOptions;
+    use std::io::Error as IOError;
+    use std::io::{Seek, SeekFrom};
+
+    type Seed = [u8; 32];
+
+    /// Account
+    ///
+    /// An account has a unique `name` and a unique `seed`,
+    /// from which the account key pair is deterministically derived.
+    #[derive(Debug, Clone)]
+    pub struct Account {
+        pub name: String,
+        pub seed: Seed,
+    }
+
+    /// StoredAccount
+    ///
+    /// The data that is stored in the filesystem relative
+    /// to an [Account]. The [Account::name] is used as the key
+    /// to this value, therefore not included here.
+    #[derive(Serialize, Deserialize)]
+    struct StoredAccount {
+        seed: Seed,
+    }
+
+    #[derive(Debug)]
+    pub enum Error {
+        /// An Account with the given name already exists
+        AlreadyExists(),
+        /// An IOError occurred
+        IOError(IOError),
+        /// A serde json Error occurred
+        JsonError(json::Error),
+    }
+
+    /// Add an account to the storage.
+    ///
+    /// Fails if an [Account] with the given name already exists.
+    /// It can also fail from IO and Serde Json errors.
+    pub fn add(account: Account) -> Result<(), Error> {
+        let mut file = get_file();
+        let mut json: HashMap<String, StoredAccount> =
+            json::from_reader(&file).ok().unwrap_or({ HashMap::new() });
+        if json.contains_key(&account.name.clone()) {
+            return Err(Error::AlreadyExists());
+        }
+
+        json.insert(account.name.clone(), StoredAccount { seed: account.seed });
+        overwrite(&mut file, &json)?;
+        Ok(())
+    }
+
+    /// List all the stored accounts
+    pub fn list() -> Vec<Account> {
+        let file = get_file();
+        let json: HashMap<String, StoredAccount> =
+            json::from_reader(&file).ok().unwrap_or({ HashMap::new() });
+        let mut accounts = Vec::new();
+        for (name, account) in json.iter() {
+            accounts.push(Account {
+                name: name.clone(),
+                seed: account.seed,
+            });
+        }
+
+        accounts
+    }
+
+    fn get_file() -> File {
+        //TODO(nuno): find a better and more qualified path.
+        let dir = BaseDirs::new().unwrap();
+        let path = dir.home_dir().join("accounts.json");
```

Comment:
On Linux this resolves to `$HOME/accounts.json` which does not seem to be a good idea. How about
```suggestion
        let path = dir.data_dir().join("radicle-registry-cli", "keys.json");
```
On linux this would resolve to `~/.local/share/radicle-registry-cli/keys.json`.

Reply:
I'm on mac and here `data_dir` translates to `$HOME/Library/Application Support` which the cli fails to create in due to permissions. I guess we want to support mac & windows?

Reply:
> I'm on mac and here `data_dir` translates to `$HOME/Library/Application Support` which the cli fails to create in due to permissions. I guess we want to support mac & windows?

Does `$HOME/Library/Application Support` not exist? If so the CLI should probably create it recursively.

Reply:
Ok, so I had seen that `write` creates the file passed if it doesn't exist so I thought it would do the same to the path's directory but apparently it doesn't. Fixed by ensuring the directory exists.